--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,26 @@
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "configure_make")
+
+filegroup(name = "all",
+          srcs = glob(["**"]),
+          visibility = ["//visibility:public"])
+
+###############
+configure_make(
+    name = "fswatch",
+    lib_source = ":all",
+    configure_in_place = True,
+    env = select({
+        "@platforms//os:macos": {
+            "AR": "",
+        },
+        "//conditions:default": {}
+    }),
+    out_static_libs = [
+        "libfswatch.a",
+    ],
+    out_shared_libs = select({
+        "@platforms//os:linux": ["libfswatch.so"],
+        "@platforms//os:macos": ["libfswatch.dylib"],
+    }),
+    visibility = ["//visibility:public"]
+)
