--- MODULE.bazel
+++ MODULE.bazel
@@ -1,8 +1,8 @@
-LIBS7_VERSION = "0.1.0"
+LIBS7_VERSION = "1.0.0"
 LIBM_S7_VERSION = "0.1.0"
 
 GOPT_VERSION = "10.0.0"
-LIBLOG_CC_VERSION = "1.0.0"
+LIBLOGC_VERSION = "1.0.0"
 PLATFORMS_VERSION = "0.0.6"
 RULES_CC_VERSION = "0.0.6"
 UNITY_VERSION = "2.5.2"
@@ -16,7 +16,7 @@
 
 bazel_dep(name = "platforms", version = PLATFORMS_VERSION)
 bazel_dep(name = "rules_cc",  version = RULES_CC_VERSION)
-bazel_dep(name = "liblog_cc",  version = LIBLOG_CC_VERSION)
+bazel_dep(name = "liblogc",  version = LIBLOGC_VERSION)
 bazel_dep(name = "libs7", version = LIBS7_VERSION)
 
 bazel_dep(name = "gopt", version = GOPT_VERSION, dev_dependency=True)
