--- MODULE.bazel
+++ MODULE.bazel
@@ -6,15 +6,15 @@
 bazel_dep(name = "platforms",    version = "0.0.6")
 bazel_dep(name = "bazel_skylib", version = "1.4.2")
 bazel_dep(name = "cc_config",    version = "1.0.0")
+bazel_dep(name = "cjson",        version = "1.7.16")
 bazel_dep(name = "liblogc",      version = "1.0.0")
+bazel_dep(name = "libs7",        version = "1.0.0")
 bazel_dep(name = "makeheaders",  version = "1.0.0")
-bazel_dep(name = "cjson",        version = "1.7.16")
-bazel_dep(name = "libs7",        version = "0.1.0")
 
 ## dev dependencies
+bazel_dep(name = "gopt",     version = "10.0.0",
+          dev_dependency=True)
 bazel_dep(name = "s7_tools", version = "1.0.0",
-          dev_dependency=True)
-bazel_dep(name = "gopt",     version = "10.0.0",
           dev_dependency=True)
 bazel_dep(name = "uthash",   version = "2.3.0",
           dev_dependency=True)
