--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,39 @@
+load("@rules_foreign_cc//foreign_cc:configure.bzl", "configure_make")
+
+filegroup(
+    name = "all",
+    srcs = glob(["**"]),
+)
+
+# build cmd:  bazel build @gdgm//:libgdbm
+configure_make(
+    name = "libgdbm",
+    lib_source = ":all",
+    out_static_libs = ["libgdbm.a"],
+    out_shared_libs = select({
+        "@platforms//os:macos": ["libgdbm.dylib"],
+        "@platforms//os:linux" : ["libgdbm.so"],
+        "//conditions:default"               : ["libgdbm.so"],
+    }),
+    env = select({
+        "@platforms//os:macos": {"AR": ""},
+        "//conditions:default": {}
+    }),
+    args = ["V=1"],
+    configure_in_place = True,
+    configure_options = [
+        "--disable-silent-rules",
+        "--disable-dependency-tracking",
+        "--without-readline",
+        ## version.c depends on __DATE__ and __TIME__, which makes the
+        ## build non-hermetic. Bazel corrects this(?) by setting both to
+        ## 'redacted', which is broken because it is not quoted.
+        ## This fixes that.
+        "CFLAGS='-Dredacted=\"redacted\"'"
+    ] + select({
+        "@platforms//os:linux" : ["--with-pic"],
+        "//conditions:default"               : []
+    }),
+    visibility = ["//visibility:public"],
+)
+
