--- MODULE.bazel
+++ MODULE.bazel
@@ -1,9 +1,7 @@
-MIBL_VERSION = "0.1.0"
-
 module(
     name = "mibl",
-    version = MIBL_VERSION,
-    compatibility_level = 0,
+    version = "1.0.0",
+    compatibility_level = 1,
 )
 
 # dev deps:
@@ -11,50 +9,29 @@
 
 bazel_dep(name = "platforms", version = "0.0.6")
 bazel_dep(name = "bazel_skylib", version = "1.4.2")
+bazel_dep(name = "cc_config",    version = "1.0.0")
 
-# rules_ocaml
-# tools_opam
-# tools_obazl
-# mibl
-
-LIBS7_VERSION = "0.1.0"
-RUNFILES_VERSION = "0.1.0"
-LIBLOG_CC_VERSION = "1.0.0"
-GOPT_VERSION = "10.0.0"
-CEXCEPTION_VERSION = "1.3.3"
-MAKEHEADERS_VERSION = "1.0.0"
-LEMON_VERSION = "3.42.0"
-RE2C_VERSION = "3.1.0"
-UNITY_VERSION = "2.5.2"
-UTHASH_VERSION = "2.3.0"
-
-CJSON_S7_VERSION = "1.7.16"
-# CJSON_VERSION = "1.7.16"
-CWALK_S7_VERSION = "1.2.7"
-# CWALK_VERSION = "1.2.7"
-DUNE_S7_VERSION = "0.1.0"
-LIBC_S7_VERSION  = "0.1.0"
-MUSTACHIOS_S7_VERSION = "0.1.0"
-TOML_S7_VERSION = "0.1.0"
-LIBTOML_CC_VERSION = "0.1.0"
-
-bazel_dep(name = "gopt", version = GOPT_VERSION)
-bazel_dep(name = "lemon", version = LEMON_VERSION)
-bazel_dep(name = "makeheaders", version = MAKEHEADERS_VERSION)
-bazel_dep(name = "re2c", version = RE2C_VERSION)
-bazel_dep(name = "uthash", version = UTHASH_VERSION)
-bazel_dep(name = "liblog_cc",  version = LIBLOG_CC_VERSION)
-bazel_dep(name = "rules_cc", version = "0.0.6")
-bazel_dep(name = "unity",  version = UNITY_VERSION, dev_dependency = True)
-
-bazel_dep(name = "runfiles", version = RUNFILES_VERSION)
-bazel_dep(name = "libs7", version = LIBS7_VERSION)
-
-bazel_dep(name = "libc_s7", version = LIBC_S7_VERSION)
-bazel_dep(name = "cwalk_s7", version = CWALK_S7_VERSION)
-bazel_dep(name = "cjson_s7", version = CJSON_S7_VERSION)
-bazel_dep(name = "dune_s7",  version = DUNE_S7_VERSION)
-bazel_dep(name = "toml_s7",  version = TOML_S7_VERSION)
-bazel_dep(name = "mustachios_s7", version = MUSTACHIOS_S7_VERSION)
+bazel_dep(name = "gopt", version = "10.0.0")
+bazel_dep(name = "lemon", version = "3.42.0")
+bazel_dep(name = "makeheaders", version = "1.0.0")
+bazel_dep(name = "re2c", version = "3.1.0")
+bazel_dep(name = "uthash", version = "2.3.0")
+bazel_dep(name = "inih",  version = "57.0.0")
+bazel_dep(name = "liblogc",  version = "1.0.0")
 
 
+bazel_dep(name = "runfiles", version = "0.1.0")
+bazel_dep(name = "libs7", version = "1.0.0")
+bazel_dep(name = "findlibc", version = "1.0.0")
+
+bazel_dep(name = "libc_s7", version = "0.1.0")
+bazel_dep(name = "libm_s7", version = "0.1.0")
+bazel_dep(name = "cwalk_s7", version = "1.2.7")
+bazel_dep(name = "cjson_s7", version = "1.7.16")
+bazel_dep(name = "dune_s7",  version = "0.1.0")
+bazel_dep(name = "toml_s7",  version = "0.1.0")
+bazel_dep(name = "mustachios_s7", version = "0.1.0")
+
+bazel_dep(name = "unity",  version = "2.5.2",
+          dev_dependency = True)
+
