--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,41 @@
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")
+
+exports_files(["libdeflate.h"])
+
+filegroup(
+    name = "all",
+    srcs = glob(["**"]),
+)
+
+cmake(
+    name = "libdeflate",
+    lib_source = ":all",
+    out_binaries = [
+        "libdeflate-gzip",
+        "libdeflate-gunzip"
+    ],
+    out_static_libs = ["libdeflate.a"],
+    out_shared_libs = select({
+        "@platforms//os:macos": ["libdeflate.dylib"],
+        "@platforms//os:linux": ["libdeflate.so"],
+        "//conditions:default": ["libdeflate.so"],
+    }),
+    # these are the defaults:
+    # cache_entries = {
+    #     "LIBDEFLATE_BUILD_STATIC_LIB": "ON",
+    #     "LIBDEFLATE_BUILD_SHARED_LIB": "ON",
+    #     "LIBDEFLATE_COMPRESSION_SUPPORT": "ON",
+    #     "LIBDEFLATE_DECOMPRESSION_SUPPORT": "ON",
+    #     "LIBDEFLATE_ZLIB_SUPPORT": "ON",
+    #     "LIBDEFLATE_GZIP_SUPPORT": "ON",
+    #     "LIBDEFLATE_FREESTANDING": "OFF",
+    #     "LIBDEFLATE_BUILD_GZIP":   "ON",
+    #     "LIBDEFLATE_BUILD_TESTS":  "OFF",
+    #     "LIBDEFLATE_USE_SHARED_LIB": "OFF"
+    # },
+    env = select({
+        "@platforms//os:macos": {"AR": "ar"},
+        "//conditions:default": {}
+    }),
+   visibility = ["//visibility:public"],
+)
