--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,34 @@
+load("@rules_foreign_cc//foreign_cc:configure.bzl",
+     "configure_make")
+
+# exports_files(["libgsl.h"])
+
+filegroup(
+    name = "all",
+    srcs = glob(["**"]),
+)
+
+# build cmd:  bazel build @libgsl//:libgsl
+configure_make(
+    name = "libgsl",
+    lib_source = ":all",
+    out_static_libs = ["libgsl.a", "libgslcblas.a"],
+    out_shared_libs = select({
+        "@platforms//os:macos": [
+            "libgsl.dylib", "libgslcblas.dylib"],
+        "@platforms//os:linux" : ["libgsl.so"],
+        "//conditions:default"               : ["libgsl.so"],
+    }),
+    env = select({
+        "@platforms//os:macos": {"AR": ""},
+        "//conditions:default": {}
+    }),
+    configure_in_place = True,
+    configure_options = [
+        "--disable-dependency-tracking",
+    ] + select({
+        "@platforms//os:linux" : ["--with-pic"],
+        "//conditions:default": []
+    }),
+    visibility = ["//visibility:public"],
+)
