--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,21 @@
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "make")
+
+filegroup(name = "all", srcs = glob(["**"]))
+
+make(
+    name = "libutf8proc",
+    lib_source = ":all",
+    env = select({
+        "@platforms//os:macos": {"AR": "ar"},
+        "//conditions:default": {}
+    }),
+    args = ["prefix=$$INSTALLDIR"],
+    out_static_libs = ["libutf8proc.a"],
+    out_shared_libs = select({
+        "@platforms//os:macos": ["libutf8proc.dylib"],
+        "@platforms//os:linux": ["libutf8proc.so"],
+        "//conditions:default": ["libutf8proc.so"],
+    }),
+    defines = ["BUILD_SHARED_LIBS"],
+    visibility = ["//visibility:public"],
+)
