--- /dev/null	2023-07-16 10:04:30
+++ cpp/BUILD.bazel	2023-07-16 10:02:41
@@ -0,0 +1,15 @@
+cc_library(
+    name  = "qrcodegen",
+    linkstatic = True,
+    srcs  = ["qrcodegen.cpp"],
+    hdrs  = ["qrcodegen.hpp"],
+    # copts = select({
+    #     "@platforms//os:linux": [
+    #         "-std=gnu11",
+    #         "-fPIC",
+    #         # "-Wl,--no-undefined",
+    #     ],
+    #     "//conditions:default": ["-std=c11"],
+    # }),
+    visibility = ["//visibility:public"]
+)
