--- /dev/null	2023-10-18 11:17:15
+++ src/BUILD.bazel	2023-10-16 08:22:59
@@ -0,0 +1,35 @@
+cc_library(
+    name  = "unity",
+    linkstatic = True,
+    srcs  = ["unity.c", "unity_internals.h"],
+    hdrs  = ["unity.h"],
+    includes = ["."],
+    copts = ["-x", "c"] + select({
+        "@platforms//os:linux": [
+            "-std=gnu11",
+            "-fPIC",
+            # "-Wl,--no-undefined",
+        ],
+        "//conditions:default": ["-std=c11"],
+    }),
+    defines = [
+        ## FIXME: parameterize with //config build settings
+        "UNITY_FLOAT_TYPE=double",
+        "UNITY_INCLUDE_DOUBLE"
+##### `UNITY_EXCLUDE_STDINT_H`
+##### `UNITY_EXCLUDE_LIMITS_H`
+##### `UNITY_INT_WIDTH`
+##### `UNITY_LONG_WIDTH`
+##### `UNITY_POINTER_WIDTH`
+##### `UNITY_SUPPORT_64`
+##### `UNITY_INCLUDE_FLOAT`
+##### `UNITY_EXCLUDE_FLOAT`
+##### `UNITY_INCLUDE_DOUBLE`
+##### `UNITY_EXCLUDE_DOUBLE`
+##### `UNITY_EXCLUDE_FLOAT_PRINT`
+##### `UNITY_FLOAT_TYPE`
+##### `UNITY_DOUBLE_TYPE`
+##### `UNITY_FLOAT_PRECISION`
+    ],
+    visibility = ["//visibility:public"]
+)
