--- /dev/null
+++ a/BUILD.bazel
@@ -0,0 +1,38 @@
+# load("//:BUILD.bzl", "BASE_COPTS")
+
+cc_library(
+    name = "cjson",
+    linkstatic = True,
+    srcs = ["cJSON.c", "cJSONx.c"],
+    hdrs = ["cJSON.h", "cJSONx.h"],
+    copts = [
+        "-x", "c",
+        "-Wall",
+        "-Wextra",
+        # GCC:
+        "-Werror", # turn all warnings into errors
+        "-Werror=pedantic", # not needed with -Werror?
+        "-Wpedantic", # same as -pedantic, strict ISO C and ISO C++ warnings
+        "-pedantic-errors",
+        "-Wfatal-errors", # stop on first error
+    ] + select({
+        # "//config/host/build:macos?": [
+        #     "-std=c11",
+        #     "-Wno-gnu-statement-expression"
+        #     # "-Werror=pedantic",
+        #     # "-Wno-gnu",
+        #     # "-Wno-format-pedantic",
+        # ],
+        "@platforms//os:linux": [
+            "-std=gnu11",
+            "-fPIC",
+            # "-Wl,--no-undefined",
+        ],
+        "//conditions:default": ["-std=c11"],
+    }),
+    # deps  = DEPS + ["//src:mustach"],
+    # linkopts = LINKOPTS,
+    # defines  = DEFINES,
+    visibility = ["//visibility:public"]
+)
+
