--- /dev/null	2023-09-02 13:32:51
+++ src/BUILD.bazel	2023-09-02 13:28:49
@@ -0,0 +1,51 @@
+exports_files(["lua.c", "lua.h", "luac.c"])
+
+filegroup(
+    name = "hdrs",
+    srcs = ["lua.h", "luaconf.h", "llimits.h",
+            "ljumptab.h", "lprefix.h"],
+    visibility = ["//bin:__pkg__", "//lib:__pkg__"]
+)
+
+filegroup(
+    name = "luac",
+    srcs = ["luac.c", "lopnames.h"],
+    visibility = ["//bin:__pkg__"]
+)
+
+CORE_O = [
+    "lapi", "lcode", "lctype", "ldebug", "ldo",
+    "lfunc", "lgc", "llex", "lmem", "lobject", "lopcodes",
+    "lparser", "lstate", "lstring", "ltable", "ltm",
+    "lundump", "lvm", "lzio"
+]
+
+filegroup(
+    name = "core_o",
+    srcs = [s + ".c" for s in CORE_O] + [s + ".h" for s in CORE_O] + ["ldump.c"],
+    visibility = ["//bin:__pkg__", "//lib:__pkg__"]
+)
+
+LIB_O = [ ## src files only
+    "lbaselib", "lcorolib", "ldblib", "liolib",
+    "lmathlib", "loadlib", "loslib", "lstrlib", "ltablib",
+    "lutf8lib", "linit"
+]
+
+filegroup(
+    name = "lib_o",
+    srcs = [s + ".c" for s in LIB_O] + [
+        "lauxlib.c", "lauxlib.h", "lualib.h"],
+    visibility = ["//bin:__pkg__", "//lib:__pkg__"]
+)
+
+################
+# experimental, one lib per object
+# cc_library(
+#     name = "lzio",
+#     srcs = [
+#         "lzio.c",
+#         "lprefix.h", "lua.h", "luaconf.h", "llimits.h",
+#         "lmem.h", "lstate.h", "lobject.h", "ltm.h", "lzio.h"
+#     ]
+# )
