--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,33 @@
+load("@rules_foreign_cc//foreign_cc:configure.bzl",
+     "configure_make")
+
+filegroup(
+    name = "all",
+    srcs = glob(["**"]),
+)
+
+configure_make(
+    name = "libmpfr",
+    lib_source = ":all",
+    deps = ["@gmp//:libgmp"],
+    linkopts = ["-lmpfr"],
+    out_static_libs = ["libmpfr.a"],
+    out_shared_libs = select({
+        "@platforms//os:macos": ["libmpfr.dylib"],
+        "@platforms//os:linux" : ["libmpfr.so"],
+        "//conditions:default"               : ["libmpfr.so"],
+    }),
+    env = select({
+        "@platforms//os:macos": {"AR": ""},
+        "//conditions:default": {}
+    }),
+    # args = ["V=1"],
+    configure_in_place = True,
+    configure_options = [
+        "--disable-maintainer-mode",
+        "--disable-silent-rules",
+        "--disable-dependency-tracking",
+    ],
+    visibility = ["//visibility:public"],
+)
+
